src/base/os.lua

```diff
@@ -80,6 +80,9 @@
 			if os.is("macosx") then
 				formats = { "lib%s.dylib", "%s.dylib" }
 				path = os.getenv("DYLD_LIBRARY_PATH") or ""
+			elseif os.is("cygwin") then
+				formats = { "lib%s.a", "%s.a", "lib%s.dll.a", "%s.dll.a" }
+				path = os.getenv("LD_LIBRARY_PATH") or ""

```

Comment:
Writing code for cygwin is much closer to linux than to windows. If you check the patch the "linux"  implementation almost always wins. So I think a cygwin should be treated as a different os.
And actually AIX, Solaris, Bsd, linux, etc, are all a unix based something with different os names.